src/contact.rs

```diff
@@ -11,64 +11,60 @@ use crate::population_loader::Alive;
 
 define_rng!(ContactRng);
 
-pub trait QueryContacts {
-    /// returns an arbitrary contact for the transmitee
-    /// a modeler in the future may institute a more complex contact model
-    /// setting weights by contact setting or shared household or the like in this method
-    fn get_contact(&mut self, transmitee: PersonId) -> Result<Option<PersonId>, IxaError>;
-    /// samples a person from a list of people with a weight by person
-    /// does not check any characteristics about the people
-    /// this is a generic method that is really just used for help sampling the person id
-    /// it is exposed to the user to allow them to make arbitrary sampling decisions
-    fn sample_person_from_list(&mut self, list: Vec<PersonId>, weights: &[f64])
-        -> Option<PersonId>;
+pub trait ContextContactExt {
+    /// Returns a potential contact for the transmitter.
+    /// Returns Ok(None) if there are no eligible contacts.
+    /// In the future, this function can be expanded to return a
+    /// contact specific to the person's household or to weight
+    /// drawing contacts be a contact matrix.
+    ///
+    /// Errors
+    /// - If there is only one person in the population.
+    fn get_contact(&mut self, transmitter_id: PersonId) -> Result<Option<PersonId>, IxaError>;
 }
 
-impl QueryContacts for Context {
-    fn get_contact(&mut self, transmitee_id: PersonId) -> Result<Option<PersonId>, IxaError> {
+#[allow(clippy::cast_possible_truncation)]
+#[allow(clippy::cast_sign_loss)]
+fn sample_person_from_list(context: &mut Context, list: &[PersonId], weights: &[f64]) -> PersonId {
+    let index = context.sample_distr(ContactRng, Categorical::new(weights).unwrap());
+    list[index as usize]
```

Comment:
```suggestion
    let index = context.sample_distr(ContactRng, Categorical::new(weights).unwrap()) as usize;
    list[index]
```

Technically speaking, the index is an integer not an f64.

Reply:
Have you checked in the doc that this is guaranteed to be floor()? Perhaps floor() first would be safer.